api/channel.go

```diff
@@ -397,10 +397,8 @@ func updateChannelPurpose(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getChannels(c *Context, w http.ResponseWriter, r *http.Request) {
-
 	// user is already in the team
 	// Get's all channels the user is a member of
-
 	if result := <-Srv.Store.Channel().GetChannels(c.TeamId, c.Session.UserId); result.Err != nil {
```

Comment:
Then just above here I think you need to add a check to make sure team ID isn't blank because that will only matter on a per API basis and can't be handled properly in context.go without having to add a new handler which I think is too much.